### PR TITLE
Biosys 112 curation

### PR DESCRIPTION
**Pull request by tony-gaia**
Many exciting changes, most notably:

* Updated edit-records-table to use Turbo Table, with all the bells and whistles that go with that.

* Added publishability, which can occur inline or in edit-record.

* Added consume step in the view records page.

* Added "SafePipe" to allow various input to be converted to "safe" input to stop the browser complaining. See:

https://stackoverflow.com/questions/38593515/warning-sanitizing-unsafe-style-value-url

* Numerous minor styling / coding improvements.

### CODE REVIEW

File: src/app/pages/view/view-records/view-records.component.ts [hunk 1]
```diff
@@ -84,21 +90,32 @@ export class ViewRecordsComponent implements OnInit {
...
         let datasetParams: any = {};
+        this.recordParams = {};
 
         if (this.projectId) {
             datasetParams['project'] = this.projectId;
         }
 
         if (this.dateStart) {
-            datasetParams['record__datetime__gte'] = this.dateStart.toISOString();
+            datasetParams['record__datetime__start'] = this.recordParams['datetime__start'] =
```
Comment (serge-gaia) [inline src/app/pages/view/view-records/view-records.component.ts]: The correct filter is now `record__datetime__gte` not 'record__datetime__start`. I've changed it last MR.

File: src/app/pages/view/view-records/view-records.component.ts [hunk 2]
```diff
@@ -84,21 +90,32 @@ export class ViewRecordsComponent implements OnInit {
...
+                this.dateStart.toISOString();
         }
 
         if (this.dateEnd) {
-            datasetParams['record__datetime__lte'] = this.dateEnd.toISOString();
+            datasetParams['record__datetime__end'] = this.recordParams['datetime__end'] =
```
Comment (serge-gaia) [inline src/app/pages/view/view-records/view-records.component.ts]: Same here use `record__datetime__lte`

File: src/app/pages/view/view-records/view-records.component.ts [hunk 3]
```diff
@@ -142,33 +159,21 @@ export class ViewRecordsComponent implements OnInit {
...
-            params['datetime__gte'] = this.dateStart.toISOString();
-        }
-
-        if (this.dateEnd) {
-            params['datetime__lte'] = this.dateEnd.toISOString();
-        }
-
-        if (this.speciesName) {
-            params['record__species_name'] = this.speciesName;
-        }
```
Comment (serge-gaia) [inline src/app/pages/view/view-records/view-records.component.ts]: This was added by me and it is needed. 
I have the feeling that you didn't pull the latest master.

File: src/app/pages/view/view-records/view-records.component.ts [hunk 4]
```diff
@@ -179,13 +184,30 @@ export class ViewRecordsComponent implements OnInit {
...
     }
 
+    public markAsConsumed() {
+        let params = JSON.parse(JSON.stringify(this.recordParams));
+
+        this.isConsuming = true;
+        this.apiService.getRecordsByDatasetId(this.selectedDataset.id, params).pipe(
+            // map((recordResponse: RecordResponse) => recordResponse.results),
+            mergeMap((records: Record[]) => from(records).pipe(
+                mergeMap((record: Record) => this.apiService.updateRecordConsumed(record.id, true))
```
Comment (serge-gaia) [inline src/app/pages/view/view-records/view-records.component.ts:206]: Is this have been merge in core? I don't think so.

**Review by serge-gaia — CHANGES_REQUESTED** [review]

File: src/app/pages/view/view-records/view-records.component.ts (hunk 1 above)
Comment (tony-gaia) [reply src/app/pages/view/view-records/view-records.component.ts]: Fixed

File: src/app/pages/view/view-records/view-records.component.ts (hunk 2 above)
Comment (tony-gaia) [reply src/app/pages/view/view-records/view-records.component.ts]: Fixed

File: src/app/pages/view/view-records/view-records.component.ts (hunk 3 above)
Comment (tony-gaia) [reply src/app/pages/view/view-records/view-records.component.ts]: I did pull the latest master down, although I might've accidentally removed this. Anyway, it's back now.

File: src/app/pages/view/view-records/view-records.component.ts (hunk 4 above)
Comment (tony-gaia) [reply src/app/pages/view/view-records/view-records.component.ts:206]: Yes

**Pull request merged**